lib/client-assets.php

```diff
@@ -371,5 +371,12 @@ function gutenberg_scripts_and_styles( $hook ) {
 		array( 'wp-components', 'wp-blocks' ),
 		filemtime( gutenberg_dir_path() . 'editor/build/style.css' )
 	);
+
+	global $hook_suffix;
+	/**
+	 * Hook to allow extending of gutenberg editor in plugin/themes.
+	 */
+	do_action( 'block_enqueue_script', $hook_suffix );
```

Comment:
I personal prefer `block_enqueue_scripts`. I'm guessing the frontend scripts and styles won't be enqueued using actions like this, but more with something tied to the block API.